### PR TITLE
Changed rules to match new http_path like `/rest?method=flickr%2Ephotos%2Esearch`

### LINKED ISSUE #945: Sysl can't parse the sysl file generated by the Swagger importer.
Please **do not post any internal, closed source snippets** on this public issue tracker!

### Description
Sysl can't parse the sysl file generated by the Swagger importer. The following fragment seems to be the problem. The fragment is generated by the swagger importer.

```
    /rest?method=flickr%2Ephotos%2Esearch:
        GET ?api_key=string&text=string?&tags=string?&user_id=string?&min_upload_date=string?&max_upload_date=string?&min_taken_date=string?&max_taken_date=string?&license=string?&sort=string?&privacy_filter=float?&bbox=string?&accuracy=string?&safe_search=float?&content_type=float?&machine_tags=string?&machine_tag_mode=string?&group_id=string?&contacts=string?&woe_id=string?&place_id=string?&media=string?&has_geo=string?&geo_context=string?&lat=string?&lon=string?&radius=float?&radius_units=string?&is_commons=bool?&in_gallery=bool?&is_getty=bool?&per_page=float?&page=float?:
            | Return a list of photos matching some criteria.
            return ok <: PhotoResource
```

### Steps to Reproduce
1. Run  sysl pb --mode json */flickr.sysl >> flickr.sysl.json
2. 
3. 

**Expected behavior:**
flickr.sysl.json created.

**Actual behavior:**
Syntax error.
line 15:9 mismatched input '?' expecting {COLON, SQ_OPEN}
line 15:16 mismatched input '=' expecting COLON
line 16:8 mismatched input 'GET ' expecting {'!wrap', '!table', '!type', '!alias', '!enum', '!union', '!view', WHATEVER, MIXIN, '<->', '.. * <- *', FORWARD_SLASH, '@', SYSL_COMMENT, TEXT_LINE, Name, E_Name}
line 16:20 mismatched input '=' expecting COLON
line 16:32 mismatched input '=' expecting COLON
line 16:45 mismatched input '=' expecting COLON
line 16:61 mismatched input '=' expecting COLON
line 16:85 mismatched input '=' expecting COLON
line 16:109 mismatched input '=' expecting COLON
line 16:132 mismatched input '=' expecting COLON
line 16:155 mismatched input '=' expecting COLON
line 16:171 mismatched input '=' expecting COLON
line 16:184 mismatched input '=' expecting COLON
line 16:207 mismatched input '=' expecting COLON
line 16:219 mismatched input '=' expecting COLON

* Debug&Crash Output

### Your Environments

```sh
$ sysl info
...

$ sysl env
...
version=0.137.0
[flickr.sysl.txt](https://github.com/anz-bank/sysl/files/4895196/flickr.sysl.txt)
[flickr.yaml.txt](https://github.com/anz-bank/sysl/files/4895197/flickr.yaml.txt)


```

### PR DESCRIPTION
**Pull request by ericzhang6222**
* Please attach **WIP** tag when this PR is still work in progress.
* Please attach **breaking-change** tag if this PR potentially causes other components to fail or changes previous sysl executable's behaviour.

Fixes #945  .

Changes proposed in this pull request:
- 
- 

Checklist:
- [x] Added related tests
- [x] Made corresponding changes to the documentation

### CODE REVIEW

**Comment by ericzhang6222** [comment]
> Are there any changes required in swagger importer to handle the change in the sysl parser? Currently query params are split.
> /rest?method=flickr%2Ephotos%2Esearch:
> GET ?api_key=string&text=string?

It can be changed in swagger importer side instead. But as definition `http_path_static: http_path_part` in `pkg/grammar/SyslParser.g4`, `/rest?method=flickr%2Ephotos%2Esearch` is static path and it is `variable` + `variable value`. It is different from `GET ?api_key=string&text=string?` which is `variable` + `variable data type`. So I changed Sysl parsing.

File: pkg/grammar/SyslParser.g4 [hunk 1]
```diff
@@ -65,7 +65,7 @@ var_in_curly    : CURLY_OPEN Name CURLY_CLOSE;
 query_var       : Name EQ (NativeDataTypes | name_str | var_in_curly) QN?;
 query_param     : QN query_var (AMP query_var)*;
 
-http_path_part :name_str | DIGITS;
+http_path_part :(name_str (QN (Name EQ Name (AMP Name EQ Name)*)+)?) | DIGITS;
```
Comment (orlade-anz) [inline pkg/grammar/SyslParser.g4:68]: Nit: Just eyeballing it, the `+` seems redundant; any subsequent param pairs would be captured as `AMP Name EQ Name`. Also space at the start :).

```suggestion
http_path_part : (name_str (QN (Name EQ Name (AMP Name EQ Name)*))?) | DIGITS;
```

File: unsorted/sysl_js/SyslParser.g4 [hunk 2]
```diff
@@ -71,7 +71,7 @@ var_in_curly    : CURLY_OPEN Name CURLY_CLOSE;
 query_var       : Name EQ (NativeDataTypes | name_str | var_in_curly) QN?;
 query_param     : QN query_var (AMP query_var)*;
 
-http_path_part :name_str | DIGITS;
+http_path_part :(name_str (QN (Name EQ Name (AMP Name EQ Name)*)+)?) | DIGITS;
```
Comment (orlade-anz) [inline unsorted/sysl_js/SyslParser.g4:74]: Same as above.
```suggestion
http_path_part : (name_str (QN (Name EQ Name (AMP Name EQ Name)*))?) | DIGITS;
```

**Comment by ericzhang6222** [comment]
@marcelocantos, can you help to review too? I changed Sysl parser rules.

**Pull request merged**